server/sandbox_run_linux.go

```diff
@@ -685,11 +709,20 @@ func (s *Server) runPodSandbox(ctx context.Context, req *types.RunPodSandboxRequ
 	if err != nil {
 		return nil, err
 	}
-	resourceCleaner.Add(func() {
-		log.Infof(ctx, "runSandbox: stopping network for sandbox %s", sb.ID())
+	description = fmt.Sprintf("runSandbox: stopping network for sandbox %s", sb.ID())
+	resourceCleaner.Add(ctx, description, func() error {
+		log.Infof(ctx, description)
 		if err2 := s.networkStop(ctx, sb); err2 != nil {
 			log.Errorf(ctx, "error stopping network on cleanup: %v", err2)
+			return err2
+		}
+		// retry with different context
+		// If we don't, this cleanup will always fail if the context times out
+		if err2 := s.networkStop(context.Background(), sb); err2 != nil {
+			log.Errorf(ctx, "error stopping network on cleanup: %v", err2)
+			return err2
 		}
+		return nil
```

Comment:
The issue with that is now if we return an error, it will cause a retry of the function. I don't think that it can work this way.